src/sentry/models/integrations/integration.py

```diff
@@ -60,6 +58,28 @@ def get_provider(self):
 
         return integrations.get(self.provider)
 
+    def delete(self, *args, **kwds):
+        with transaction.atomic(), test_psql_role_override("postgres"):
+            for outbox in Integration.outboxes_for_update(self.id):
+                outbox.save()
+            return super().delete(*args, **kwds)
+
+    @staticmethod
+    def outboxes_for_update(identifier: int) -> List[ControlOutbox]:
+        org_ids: List[int] = OrganizationIntegration.objects.filter(
```

Comment:
ControlOutboxes, unlike RegionOutboxes, are multiplexed -- they are sent to multiple destinations.  For this reason, I general make helpers associated with outbox creation when it is a Control Silo model, to ensure the logic of generating all the required outboxes is correct.

In this case, we need to replicate integration deletion to all regions that belong to all `OrganizationIntegration` entries associated.

Reply:
Maybe not necessary to do, but should we assert this exists? Could the @control_silo_model decorate check for hybrid cloud FKs that cascade and assert an `outboxes_for_update` method exists?